approvals/approvals/api.py

```diff
@@ -191,51 +209,62 @@ def assign_approvers(doc: Document, method: str | None = None):
 
 
 @frappe.whitelist()
-def add_user_approval(doc: Document, method: str | None = None, user=None):
+def add_user_approval(doc: Document | str, method: str | None = None, user=None):
 	if not user:
 		return
-	doc = frappe._dict(json.loads(doc)) if isinstance(doc, str) else doc
+	doc = frappe.get_doc(json.loads(doc)) if isinstance(doc, str) else doc
 	uda = frappe.new_doc("User Document Approval")
 	uda.reference_doctype = doc.doctype
 	uda.reference_name = doc.name
 	uda.approver = user
 	uda.save(ignore_permissions=True)
-	frappe.db.commit()
+
+	add_approval_comment(
+		doc.doctype,
+		doc.name,
+		f"<b>{user}<b> added as approver by <b>{frappe.session.user}</b>",
+		subject="Approver added",
+		comment_type="Info",
+	)
 
 
 @frappe.whitelist()
-def remove_user_approval(doc: Document, method: str | None = None, user=None):
-	doc = frappe._dict(json.loads(doc)) if isinstance(doc, str) else doc
+def remove_user_approval(doc: Document | str, method: str | None = None, user=None):
+	doc = frappe.get_doc(json.loads(doc)) if isinstance(doc, str) else doc
 	user_approval = frappe.get_doc(
 		"User Document Approval",
 		{"reference_doctype": doc.doctype, "reference_name": doc.name, "approver": user},
 	)
 	user_approval.delete()
-	removal = frappe.new_doc("Comment")
-	removal.reference_doctype = doc.doctype
-	removal.reference_name = doc.name
-	removal.comment_type = "Info"
-	removal.comment_email = frappe.session.user
-	removal.content = f"<b>{user}<b> removed as approver by <b>{frappe.session.user}</b>"
-	removal.subject = "Approver removed"
-	removal.save(ignore_permissions=True)
-	return
+
+	add_approval_comment(
+		doc.doctype,
+		doc.name,
+		f"<b>{user}<b> removed as approver by <b>{frappe.session.user}</b>",
+		subject="Approver removed",
+		comment_type="Info",
+	)
 
 
 @frappe.whitelist()
-def create_approval_notification(doc: Document, user):
-	no = frappe.new_doc("Notification Log")
-	no.flags.ignore_permissions = True
-	no.owner = "Administrator"
-	no.for_user = user
-	no.subject = f"A {doc.doctype} requires your approval"
-	no.type = "Assignment"
-	no.document_type = doc.doctype
-	no.document_name = doc.name
-	no.from_user = doc.owner
-	no.email_content = f"{doc.doctype} {doc.name} requires your approval"
+def create_approval_notification(doc: Document | frappe._dict, user):
```

Comment:
@agritheory I've avoided creating a comment around calls for this function since assigning automatically creates a comment in the timeline.